cpp/include/raft/cluster/single_linkage_types.hpp

```diff
@@ -27,23 +29,33 @@ enum LinkageDistance { PAIRWISE = 0, KNN_GRAPH = 1 };
  * @tparam value_idx
  * @tparam value_t
  */
-template <typename value_idx, typename value_t>
+template <typename idx_t>
 class linkage_output {
  public:
-  value_idx m;
-  value_idx n_clusters;
+  idx_t m;
+  idx_t n_clusters;
+
+  idx_t n_leaves;
+  idx_t n_connected_components;
 
-  value_idx n_leaves;
-  value_idx n_connected_components;
+  // TODO: These will be made private in a future release
+  idx_t* labels;    // size: m
```

Comment:
[optional] Would you consider just storing these two arrays as `device_{matrix,vector}_view`?  Also, if `get_children()` always returns an mdspan with 2 columns, would you consider encoding that in the returned mdspan's `extents` type (e.g., `extents<idx_t, dynamic_extent, 2>`)?

Reply:
I'd actually like to remove this class in a future version and just accept the outputs directly. I can't remember why this was necessary but it was long before  we started focusing on end-user consumption.

Reply:
(As for the reason I just added the mdspan getters in the meantime, I was trying not to break the existing API right away.)>